src/org/zaproxy/zap/extension/groovy/GroovyEngineWrapper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Zed Attack Proxy (ZAP) and its related class files.
+ *
+ *  ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ *  Copyright 2018 The ZAP Development Team
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.groovy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineFactory;
+import javax.swing.ImageIcon;
+
+import org.codehaus.groovy.jsr223.GroovyScriptEngineFactory;
+import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
+import org.zaproxy.zap.control.ExtensionFactory;
+import org.zaproxy.zap.extension.script.DefaultEngineWrapper;
+
+public class GroovyEngineWrapper extends DefaultEngineWrapper {
+
+	private final ScriptEngineFactory factory;
+
+	public static ScriptEngine getScriptEngineFromFactory()
+	{
+		return getScriptEngineFromFactory(new GroovyScriptEngineFactory());
+	}
+
+	private static ScriptEngine getScriptEngineFromFactory(ScriptEngineFactory factory){
+		ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+		Thread.currentThread().setContextClassLoader(new AddOnClassLoaderWrapper(ExtensionFactory.getAddOnLoader()));
+		try {
+			return factory.getScriptEngine();
+		} finally {
+			Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+		}
+	}
+
+	public GroovyEngineWrapper(ScriptEngine engine) {
```

Comment:
This could be constructed directly with the `GroovyScriptEngineFactory`, it's not necessary to wrap the engine with a custom class loader here.